src/wallet/spend.cpp

```diff
@@ -1294,6 +1294,11 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
         return util::Error{TransactionErrorString(TransactionError::MAX_FEE_EXCEEDED)};
     }
 
+    CFeeRate tx_fee_rate = CFeeRate(current_fee, nBytes);
+    if (tx_fee_rate > wallet.m_max_tx_fee_rate) {
+        return util::Error{TransactionErrorString(TransactionError::MAX_FEE_EXCEEDED)};
```

Comment:
perhaps introduce a new error type: 

```suggestion
        return util::Error{TransactionErrorString(TransactionError::MAX_FEE_RATE_EXCEEDED)};
```

Reply:
Added

Reply:
I realized now that you are adding the new Error type in the third PR. It would make more sense to me if the new Error type were introduced first, and then the test that uses it would get added rather than adding the test and then later introducing the new Error type.

Reply:
Thanks @murchandamus, I agree with your suggestion and reorder the commits.